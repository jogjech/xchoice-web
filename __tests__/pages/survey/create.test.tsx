import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import CreatePage from "../../../pages/survey/create";

describe("Survey page", () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<CreatePage></CreatePage>);
  });

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("contains layout", () => {
    const layout = wrapper.find("Layout");
    expect(layout.exists()).toBeTruthy();
    expect(layout.prop("title")).toBe("Create Survey");
  });

  it("contains survey creator", () => {
    const surveyCreator = wrapper.find("SurveyCreator");
    expect(surveyCreator.exists()).toBeTruthy();
  });
});
