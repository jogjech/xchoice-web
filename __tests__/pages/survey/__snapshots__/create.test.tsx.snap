// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Survey page renders correctly 1`] = `
<Layout
  requiredAuth={true}
  title="Create Survey"
>
  <Memo(BackTop) />
  <SurveyCreator />
</Layout>
`;
